services/storage/src/simcore_service_storage/dsm_factory.py

```diff
@@ -50,7 +51,12 @@ async def list_files_in_dataset(
 
     @abstractmethod
     async def list_files(
-        self, user_id: UserID, *, expand_dirs: bool, uuid_filter: str = ""
+        self,
+        user_id: UserID,
+        *,
+        expand_dirs: bool,
+        uuid_filter: str = "",
+        project_id: ProjectID | None = None,
```

Comment:
and question: since the handler should actually set the uuid_filter/project_id fields. does it make sense to have default arguments?

Reply:
I think it makes sense to have default values here as well because some of these `list_file` methods are also called within the code. Both the `uuid_filter` and `project_id` are filtering parameters, and the defaults are chosen in such away that those filters are disabled if they are not passed. That makes sense to me, but let me know if you disagree.

Reply:
Good point. I have removed the defaults from the service/business logic layer.